core/services/flux_monitor.go

```diff
@@ -309,18 +316,39 @@ func (p *PollingDeviationChecker) Start(ctx context.Context, client eth.Client)
 	return nil
 }
 
+// stopTimer stops and clears the timer as suggested by the documentation.
+func stopTimer(arg *time.Timer) {
+	if !arg.Stop() && len(arg.C) > 0 {
+		// timer's channel may have backlog
+		// refer to: https://developpaper.com/detailed-explanation-of-the-trap-of-timer-in-golang/
+		<-arg.C
+	}
+}
+
 func (p *PollingDeviationChecker) consume(ctx context.Context, roundSubscription eth.Subscription) {
 	defer roundSubscription.Unsubscribe()
+
+	idleThreshold := time.NewTimer(math.MaxInt64)
+	defer stopTimer(idleThreshold)
+
 	for {
+		if p.idleThreshold > 0 {
+			stopTimer(idleThreshold) // Reset expects stopped or expired timer.
+			idleThreshold.Reset(p.idleThreshold)
+		}
+
 		select {
 		case <-ctx.Done():
+			close(p.stopC)
 			return
 		case err := <-roundSubscription.Err():
 			logger.Error(errors.Wrap(err, "checker lost subscription to NewRound log events"))
```

Comment:
I personally cannot say.  @se3000 should we create a follow up task for it?  Has one been created?

Reply:
@felder-cl You can if you'd like to. It seems like it's in the scope of this story if it's something you'd like to simplify/refactor. Your call, maybe do it now but as a separate PR? We should definitely put some time into it one way or another. I believe the method is relatively new, and we could probably delete some code on our end if it works.

Reply:
I've opted for its own story.  Given it would probably require its own test/s etc., it could be a significant delay.

https://www.pivotaltracker.com/story/show/171234713